### PR TITLE
NAS-118068 / 22.12 / add R50BM to enclosure mapping code

### PR DESCRIPTION
**Pull request by yocalebo**
The R50BM uses the same plx bridge as our M50/60 series so instead of writing, literally, a duplicate function I've refactored the code that does all the rear nvme drive bay mapping into a singular api call `enclosure.map_nvme`.

Fixes:
1. refactor to 1 file `enclosure_/nvme.py` for all rear nvme bay drive mapping logic
2. add R50BM enclosure mapping support

### CODE REVIEW

**Comment by yocalebo** [comment]
retest this please

**Comment by yocalebo** [comment]
retest this please

**Comment by yocalebo** [comment]
retest this please

**Comment by yocalebo** [comment]
retest this please

**Comment by yocalebo** [comment]
retest this please

File: src/middlewared/middlewared/pytest/unit/plugins/enclosure/test_m_series_nvme.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def test__m_series_nvme_enclosures(fs):
             )
         )
 
-        EnclosureService(middleware).m_series_nvme_enclosures()\
+        EnclosureService(middleware).map_nvme()\
```
Comment (themylogin) [inline src/middlewared/middlewared/pytest/unit/plugins/enclosure/test_m_series_nvme.py:45]: It was a typo
```suggestion
        EnclosureService(middleware).map_nvme()
```

**Review by themylogin — APPROVED** [review]

File: src/middlewared/middlewared/pytest/unit/plugins/enclosure/test_m_series_nvme.py (hunk 1 above)
Comment (yocalebo) [reply src/middlewared/middlewared/pytest/unit/plugins/enclosure/test_m_series_nvme.py:45]: ahh, okay. I saw it but thought maybe something specific for pytest or something. Thanks!

**Pull request merged**